cellprofiler/image.py

```diff
@@ -56,162 +39,55 @@ class Image(object):
-        elif img.dtype.type is np.uint16:
-            scale = math.pow(2.0, 16.0) - 1
-        elif img.dtype.type is np.uint32:
-            scale = math.pow(2.0, 32.0) - 1
-        elif img.dtype.type is np.uint64:
-            scale = math.pow(2.0, 64.0) - 1
-        elif img.dtype.type is np.int8:
-            scale = math.pow(2.0, 8.0)
-            mval = -scale / 2.0
-            scale -= 1
-            fix_range = True
-        elif img.dtype.type is np.int16:
-            scale = math.pow(2.0, 16.0)
-            mval = -scale / 2.0
-            scale -= 1
-            fix_range = True
-        elif img.dtype.type is np.int32:
-            scale = math.pow(2.0, 32.0)
-            mval = -scale / 2.0
-            scale -= 1
-            fix_range = True
-        elif img.dtype.type is np.int64:
-            scale = math.pow(2.0, 64.0)
-            mval = -scale / 2.0
-            scale -= 1
-            fix_range = True
-        # Avoid temporaries by doing the shift/scale in place.
-        img = img.astype(np.float32)
-        img -= mval
-        img /= scale
-        if fix_range:
-            # These types will always have ranges between 0 and 1. Make it so.
-            np.clip(img, 0, 1, out=img)
-        check_consistency(img, self.__mask)
-        self.__image = ImageCache(img)
-
-    image = property(get_image, set_image)
-    pixel_data = property(get_image, set_image)
-
-    def get_parent_image(self):
-        """The image from which this one was derived"""
-        return self.__parent_image
-
-    def set_parent_image(self, parent_image):
-        self.__parent_image = parent_image
-
-    parent_image = property(get_parent_image, set_parent_image)
-
-    def get_has_parent_image(self):
-        """True if this image has a defined parent"""
-        return self.__parent_image is not None
-
-    has_parent_image = property(get_has_parent_image)
-
-    def get_masking_objects(self):
-        """The objects used to crop and mask this image"""
-        return self.__masking_objects
-
-    def set_masking_objects(self, value):
-        self.__masking_objects = value
-
-    masking_objects = property(get_masking_objects, set_masking_objects)
-
-    def get_has_masking_objects(self):
-        """True if the image was cropped with objects
-
-        If this is true, there will also be a valid labels matrix
-        available through the labels property
-        """
-        return self.__masking_objects is not None
+        # self.filename = filename
+        self.pathname = pathname
+        self.channel_names = None
+        self.has_parent_image = self.parent is not None
+        self.has_masking_objects = self.__masking_objects is not None
+        self.labels = self.crop_image_similarly(self.masking_objects.segmented) if self.has_masking_objects else None
+        self.has_channel_names = self.channel_names is not None
+        self.scale = self.parent.scale if self.__scale is None and self.has_parent_image else self.__scale
```

Comment:
`self.scale` can probably be safely removed since it’s only used once (inside `cellprofiler.modules.LoadImages`)